### DISCUSSION TITLE
Decoupling Task Scheduling and Execution Across Two Applications

### DISCUSSION THREAD

**Discussion by KBA2024** [kagkarlsson/db-scheduler#568] [Q&A] [answered]
I have a use case where I need to decouple task scheduling and execution between two separate applications:

App 1: Responsible only for scheduling tasks (e.g., daily, every 5 minutes, or immediate).
App 2: Responsible for running the tasks without being aware of the scheduling logic or frequency (this is outside its scope).
Is it possible to achieve this separation of concerns using db-scheduler? If so, what would be the best approach to implement this architecture?

**Comment by kagkarlsson** [comment, marked as answer]
It is possible to split scheduling and execution, but not without the executor knowing anything about the schedule or frequency. In db-scheduler, the next execution-time is calculated when the current execution is done, and to calculate that it needs access to the schedule. One variant is what I called dynamic recurring tasks, where the schedule is stored in the `task_data` rather than statically in code ([an example](https://github.com/kagkarlsson/db-scheduler/blob/master/examples/features/src/main/java/com/github/kagkarlsson/examples/RecurringTaskWithPersistentScheduleMain.java))

Normally, when splitting scheduling and execution, you would start the Scheduler in the executor-application and only instantiate a SchedulerClient in the scheduling-application.

Appreciate the donation 🙏
